plugin/storage/es/factory.go

```diff
@@ -276,13 +280,36 @@ func createSpanWriter(
 		if err != nil {
 			return nil, err
 		}
-		if err := writer.CreateTemplates(spanMapping, serviceMapping, cfg.Indices.IndexPrefix); err != nil {
+		if err := CreateTemplatesWithRetry(ctx, writer, spanMapping, serviceMapping, cfg.Indices.IndexPrefix, logger); err != nil {
 			return nil, err
 		}
 	}
 	return writer, nil
 }
 
+func CreateTemplatesWithRetry(
+	ctx context.Context,
+	writer *esSpanStore.SpanWriter,
+	spanMapping string,
+	serviceMapping string,
+	indexPrefix cfg.IndexPrefix,
+	logger *zap.Logger,
+) error {
+	for i := 0; i < 3; i++ {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("template creation timeout: %w", ctx.Err())
+		default:
+			if err := writer.CreateTemplates(ctx, spanMapping, serviceMapping, indexPrefix); err == nil {
```

Comment:
I am not seeing how this addresses the original issue. The assumption was that when we call CreateTemplates it may return before the templates are actually created in the ES, and the immediately-after span writes might fail. Doing retries does not address that.

I would also like to understand if that assumption is correct in the first place and why it might happen - is it indeed that ES acks a success before the template changes are actually applied to the db? Or could it be our fault, e.g. by using an async connection? We do something funky with batch inserts that happens in the background, that could be the source of a race.